instrumentation/grizzly-2.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/grizzly/GrizzlyFilterchainServerTest.java

```diff
@@ -0,0 +1,235 @@
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint;
+import io.opentelemetry.semconv.HttpAttributes;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import org.glassfish.grizzly.Transport;
+import org.glassfish.grizzly.filterchain.BaseFilter;
+import org.glassfish.grizzly.filterchain.FilterChain;
+import org.glassfish.grizzly.filterchain.FilterChainBuilder;
+import org.glassfish.grizzly.filterchain.FilterChainContext;
+import org.glassfish.grizzly.filterchain.NextAction;
+import org.glassfish.grizzly.filterchain.TransportFilter;
+import org.glassfish.grizzly.http.HttpContent;
+import org.glassfish.grizzly.http.HttpHeader;
+import org.glassfish.grizzly.http.HttpRequestPacket;
+import org.glassfish.grizzly.http.HttpResponsePacket;
+import org.glassfish.grizzly.http.HttpServerFilter;
+import org.glassfish.grizzly.http.util.Parameters;
+import org.glassfish.grizzly.nio.transport.TCPNIOTransport;
+import org.glassfish.grizzly.nio.transport.TCPNIOTransportBuilder;
+import org.glassfish.grizzly.utils.DelayedExecutor;
+import org.glassfish.grizzly.utils.IdleTimeoutFilter;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class GrizzlyFilterchainServerTest extends AbstractHttpServerTest<Transport> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
+
+  private TCPNIOTransport transport;
+
+  @Override
+  protected Transport setupServer() throws Exception {
+    FilterChain filterChain = setUpFilterChain();
+    setUpTransport(filterChain);
+
+    transport.bind("127.0.0.1", port);
+    transport.start();
+    return transport;
+  }
+
+  @Override
+  protected void stopServer(Transport transport) throws IOException {
+    transport.stop();
+  }
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    options.setHttpAttributes(
+        serverEndpoint -> {
+          Set<AttributeKey<?>> attributes =
+              new HashSet<>(HttpServerTestOptions.DEFAULT_HTTP_ATTRIBUTES);
+          attributes.remove(HttpAttributes.HTTP_ROUTE);
+          return attributes;
+        });
+
+    // justification: grizzly async closes the channel which
+    // looks like a ConnectException to the client when this happens
+    options.setTestException(false);
+
+    options.setTestCaptureHttpHeaders(false);
+    options.setVerifyServerSpanEndTime(
+        false); // server spans are ended inside of the controller spans
```

Comment:
the formatter did some odd stuff to some of the comments

```suggestion
    // server spans are ended inside of the controller spans
    options.setVerifyServerSpanEndTime(false); 
```